File: src/main/java-graphql/io/github/graphql/java/resolver/GistResolver.java
package io.github.graphql.java.resolver;

import io.github.graphql.java.model.*;

/**
 * Resolver for Gist
 */
@javax.annotation.Generated(
    value = "com.kobylynskyi.graphql.codegen.GraphQLCodegen",
    date = "2021-03-07T19:46:14+0800"
)
public interface GistResolver {

    @javax.validation.constraints.NotNull
    GistCommentConnectionTO comments(GistTO gistTO, String after, String before, Integer first, Integer last) throws Exception;

    java.util.List<GistFileTO> files(GistTO gistTO, Integer limit, String oid) throws Exception;

    @javax.validation.constraints.NotNull
    GistConnectionTO forks(GistTO gistTO, String after, String before, Integer first, Integer last, GistOrderTO orderBy) throws Exception;

    @javax.validation.constraints.NotNull
    StargazerConnectionTO stargazers(GistTO gistTO, String after, String before, Integer first, Integer last, StarOrderTO orderBy) throws Exception;

}

File: src/main/kotlin-graphql/io/github/graphql/resolver/GistResolver.kt
package io.github.graphql.resolver

import io.github.graphql.model.*

/**
 * Resolver for Gist
 */
@javax.annotation.Generated(
    value = ["com.kobylynskyi.graphql.codegen.GraphQLCodegen"],
    date = "2021-03-06T19:48:04+0800"
)
interface GistResolver {

    @Throws(Exception::class)
    fun comments(gistTO: GistTO, after: String?, before: String?, first: Int?, last: Int?): GistCommentConnectionTO

    @Throws(Exception::class)
    fun files(gistTO: GistTO, limit: Int?, oid: String?): List<GistFileTO?>?

    @Throws(Exception::class)
    fun forks(gistTO: GistTO, after: String?, before: String?, first: Int?, last: Int?, orderBy: GistOrderTO?): GistConnectionTO

    @Throws(Exception::class)
    fun stargazers(gistTO: GistTO, after: String?, before: String?, first: Int?, last: Int?, orderBy: StarOrderTO?): StargazerConnectionTO

}
